src/fs.zig

```diff
@@ -1352,63 +1352,15 @@ pub const NodeJSPathName = struct {
 
     pub fn init(_path: string, comptime isWindows: bool) NodeJSPathName {
         const platform: path_handler.Platform = if (isWindows) .windows else .posix;
-        const getLastSep = comptime platform.getLastSeparatorFunc();
-
-        var path = _path;
-        var base = path;
-        // ext must be empty if not detected
-        var ext: string = "";
-        var dir = path;
-        var is_absolute = true;
-        var _i = getLastSep(path);
-        var first = true;
-        while (_i) |i| {
-            // Stop if we found a non-trailing slash
-            if (i + 1 != path.len and path.len >= i + 1) {
-                base = path[i + 1 ..];
-                dir = path[0..i];
-                is_absolute = false;
-                break;
-            }
-
-            // If the path starts with a slash and it's the only slash, it's absolute
-            if (i == 0 and first) {
-                base = path[1..];
-                dir = &([_]u8{});
-                break;
-            }
-
-            first = false;
-            // Ignore trailing slashes
-
-            path = path[0..i];
-
-            _i = getLastSep(path);
-        }
-
-        // clean trailing slashs
-        if (base.len > 1 and platform.isSeparator(base[base.len - 1])) {
-            base = base[0 .. base.len - 1];
-        }
-
+        const isAbsolute = platform.isAbsolute(_path);
+        const dir = if (isAbsolute) &([_]u8{}) else platform.dirname(_path) orelse &([_]u8{});
+        const base = platform.basename(_path);
+        const ext = std.fs.path.extension(base);
         // filename is base without extension
         var filename = base;
-
-        // if only one character ext = "" even if filename it's "."
-        if (filename.len > 1) {
-            // Strip off the extension
-            if (strings.lastIndexOfChar(filename, '.')) |dot| {
-                if (dot > 0) {
-                    filename = filename[0..dot];
-                    ext = base[dot..];
-                }
-            }
+        if (filename.len > 1 and ext.len > 0) {
+            filename = filename[0 .. filename.len - ext.len];
         }
-
-        if (is_absolute) {
-            dir = &([_]u8{});
-        }
-
```

Comment:
☝️ The deleted code above looked to be doing what the standard zig lib does. So I replaced it with the std lib to reduce gotchas. I kept the deviation in behavior though. I need a second look here though 🙏 .